File: i2f-spring/i2f-spring-core/src/main/java/i2f/spring/matcher/MatcherUtil.java
package i2f.spring.matcher;

import org.springframework.util.AntPathMatcher;
import org.springframework.util.StringUtils;

import java.util.Collection;

/**
 * @author Ice2Faith
 * @date 2023/6/13 9:52
 * @desc
 */
public class MatcherUtil {
    private static AntPathMatcher antUrlMatcher = new AntPathMatcher("/");
    private static AntPathMatcher antPkgMatcher = new AntPathMatcher(".");

    public static boolean antUrlMatched(String str, String patten) {
        return antUrlMatcher.match(patten, str);
    }

    public static boolean antPkgMatched(String str, String patten) {
        return antPkgMatcher.match(patten, str);
    }

    public static boolean antUrlMatchedAny(String str, Collection<String> pattens) {
        for (String patten : pattens) {
            if (StringUtils.isEmpty(patten)) {
                continue;
            }
            if (antUrlMatched(str, patten)) {
                return true;
            }
        }
        return false;
    }

    public static boolean antPkgMatchedAny(String str, Collection<String> pattens) {
        for (String patten : pattens) {
            if (StringUtils.isEmpty(patten)) {
                continue;
            }
            if (antPkgMatched(str, patten)) {
                return true;
            }
        }
        return false;
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-core/match/MatcherUtil.js
import AntMatcher from "./impl/AntMatcher";

const MatcherUtil = {
    /**
     * @type {IMatcher}
     */
    antUrlMatcher: new AntMatcher("/"),
    /**
     * @type {IMatcher}
     */
    antPkgMatcher: new AntMatcher("."),
    /**
     *
     * @param str {String}
     * @param patten {String}
     * @return {boolean}
     */
    antUrlMatched(str, patten) {
        return this.antUrlMatcher.isMatch(str, patten);
    },
    /**
     *
     * @param str {String}
     * @param patten {String}
     * @return {boolean}
     */
    antPkgMatched(str, patten) {
        return this.antPkgMatcher.isMatch(str, patten);
    },
    /**
     *
     * @param str {String}
     * @param pattens {string[]}
     * @return {boolean}
     */
    antUrlMatchedAny(str, pattens) {
        for (let i = 0; i < pattens.length; i++) {
            let patten = pattens[i]
            if (!patten) {
                continue;
            }
            if (this.antUrlMatched(str, patten)) {
                return true;
            }
        }
        return false;
    },
    /**
     *
     * @param str {String}
     * @param pattens {string[]}
     * @return {boolean}
     */
    antPkgMatchedAny(str, pattens) {
        for (let i = 0; i < pattens.length; i++) {
            let patten = pattens[i]
            if (!patten) {
                continue;
            }
            if (antPkgMatched(str, patten)) {
                return true;
            }
        }
        return false;
    }
}

export default MatcherUtil